Src/FluentAssertions/Common/TypeMemberReflector.cs

```diff
@@ -31,27 +31,39 @@ private static PropertyInfo[] LoadNonPrivateProperties(Type typeToReflect, Membe
     {
         IEnumerable<PropertyInfo> query =
             from propertyInfo in GetPropertiesFromHierarchy(typeToReflect, visibility)
-            where HasNonPrivateGetter(propertyInfo)
+            where HasNonPrivateGetter(propertyInfo, visibility)
             where !propertyInfo.IsIndexer()
             select propertyInfo;
 
         return query.ToArray();
     }
 
-    private static List<PropertyInfo> GetPropertiesFromHierarchy(Type typeToReflect, MemberVisibility memberVisibility)
+    private static IEnumerable<PropertyInfo> GetPropertiesFromHierarchy(Type typeToReflect, MemberVisibility memberVisibility)
     {
-        bool includeInternals = memberVisibility.HasFlag(MemberVisibility.Internal);
+        bool includeInternal = memberVisibility.HasFlag(MemberVisibility.Internal);
 
         return GetMembersFromHierarchy(typeToReflect, type =>
         {
             return type
                 .GetProperties(AllInstanceMembersFlag | BindingFlags.DeclaredOnly)
-                .Where(property => property.GetMethod?.IsPrivate == false)
```

Comment:
This also filter out properties without a getter.
How about we change it to just check if `property.GetMethod` is not null?

Reply:
Are you saying that we _should_ filter out properties without a getter?

Reply:
What would be the added value?

Reply:
If `includeInternal` is true, we can now return a member without a getter, whereas currently `Where(property => property.GetMethod?.IsPrivate == false)` requires the property to have a getter.

But I see this doesn't matter in practice since `HasGetter` will discard any property without a getter .
![image](https://github.com/fluentassertions/fluentassertions/assets/919634/f0f2ae06-3912-48a7-9dc8-bfc9902b604b)

So I withdraw my comments on this one.